### PR TITLE
[SPARK-42254][SQL] Assign name to _LEGACY_ERROR_TEMP_1117

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes to assign name to _LEGACY_ERROR_TEMP_1117, "REQUIRES_SINGLE_PART_NAMESPACE".

### Why are the changes needed?

We should assign proper name to _LEGACY_ERROR_TEMP_*

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

`./build/sbt "sql/testOnly org.apache.spark.sql.SQLQueryTestSuite*`

### CODE REVIEW

**Comment by itholic** [comment]
cc @srielau @MaxGekk @cloud-fan could you review this error class when you find some time??

File: core/src/main/resources/error/error-classes.json [hunk 1]
```diff
@@ -1208,6 +1208,11 @@
     ],
     "sqlState" : "42K03"
   },
+  "REQUIRES_SINGLE_PART_NAMESPACE" : {
+    "message" : [
+      "<sessionCatalog> requires a single-part namespace, but got <namespace>."
+    ]
```
Comment (srielau) [inline core/src/main/resources/error/error-classes.json:1214]:
```suggestion
    ],
    "sqlState" : "42K05"
```

**Review by srielau — APPROVED** [review]

File: core/src/main/resources/error/error-classes.json (hunk 1 above)
Comment (itholic) [reply core/src/main/resources/error/error-classes.json]: Updated, thanks!

**Review by LuciferYang — APPROVED** [review]
+1, LGTM

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala [hunk 2]
```diff
@@ -1239,12 +1239,12 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase {
...
 
-  def requiresSinglePartNamespaceError(ns: Seq[String]): Throwable = {
+  def requiresSinglePartNamespaceError(namespace: Seq[String]): Throwable = {
     new AnalysisException(
-      errorClass = "_LEGACY_ERROR_TEMP_1117",
+      errorClass = "REQUIRES_SINGLE_PART_NAMESPACE",
       messageParameters = Map(
         "sessionCatalog" -> CatalogManager.SESSION_CATALOG_NAME,
-        "ns" -> ns.mkString("[", ", ", "]")))
+        "namespace" -> namespace.mkString(".")))
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala:1247]: Please, quote the namespace id:
```suggestion
        "namespace" -> toSQLId(namespace)))
```

**Review by MaxGekk — CHANGES_REQUESTED** [review]

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala (hunk 2 above)
Comment (itholic) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala]: Updated, thanks!

**Review by MaxGekk — APPROVED** [review]
Waiting for CI.

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master/3.4.
Thank you, @itholic and @LuciferYang @srielau for review.

**Pull request closed without merging**